### DISCUSSION TITLE
How to auto copy the selected word to the search input bar?

### DISCUSSION THREAD

**Discussion by ghost** [vscode-neovim/vscode-neovim#1827] [Q&A]
This is my mapping
```
  xnoremap <Space>w <Cmd>call VSCodeNotify('search.action.openNewEditor')<CR><Esc>
```
Say I first visually select a word, and then run the above mapping.  Now, a new search editor will be open,  however, the previously selected word is not shown in the search editor bar.  I'd expect the word will show there by default.

**Comment by xiyaowong** [comment]
Pass [args](https://github.com/microsoft/vscode/blob/eb278bd53bf8c7f67de33e8bd1a3a7725edfc010/src/vs/workbench/contrib/searchEditor/browser/searchEditor.contribution.ts#L179) to the command.
```lua
vim.keymap.set(
  'x',
  '<Space>w',
  function()
    require('vscode-neovim').action('search.action.openNewEditor', {
      args = {
        query = 'abc',
      },
    })
  end
)
```

**Reply by ghost** [reply]
The 'abc' is a static word. Do you know how to pass the current selected word?

**Reply by xiyaowong** [reply]
`:h nvim_buf_get_text`
